system/lib/libc/musl/src/legacy/isastream.c

```diff
@@ -1,7 +1,19 @@
+#ifdef __EMSCRIPTEN__
+#include <errno.h>
+#endif
 #include <stropts.h>
 #include <fcntl.h>
 
 int isastream(int fd)
```

Comment:
```The isastream() function shall test whether fildes, an open file descriptor, is associated with a STREAMS-based file.```

I'm not really sure that this means.   It seems different to "is this a valid file descriptor".  I'm tempted not to include legacy code but rather to write a new function with a more meaningful name.   e.g. `__wasi_fd_is_valid`  (assuming that is the intent?)

Reply:
Yeah, the docs may be a little confusing, but from the code it seems like STREAMS is just "any file" (at least in modern systems), so it checks if an fd is valid.

But makes sense to use a special new helper, I added `__wasi_fd_is_valid` now, I agree that might be cleaner.

Reply:
How about we simply leave this file unchanged?

Reply:
A downside to leaving it unchanged is we have more unconverted stuff, so a grep for unconverted stuff will find this.

What's the downside to changing it, in your opinion?

Reply:
It just doesn't make sense to me to carry patches to code we never build.  Better to delete the file completely maybe?   I'm not sure.

I don't feel strongly about it.  You can land this as is if you like.

Reply:
Fair enough, I'll revert that part.